### DISCUSSION TITLE
Solar overview on display

### DISCUSSION THREAD

**Discussion by TheSerapher** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#115] [Show and tell]
Hey chaps,

Always wanted to use my box as an overview display for my solar production and finally found the time to look into it. After some fiddling around and pushing pixels left and right I am quite happy with the result!

![IMG_0119](https://github.com/user-attachments/assets/e38d7a61-6f7c-4542-a749-d19939d397dd)

I have edited the original code quite heavily to clean things up for my use case so it may not be something for you. To make things available for other to copy and modify to their hearts content, here a Gist of it:

* solar overview: https://gist.github.com/TheSerapher/2602ca9ef656c2818b54b158a65da9e5#file-gistfile1-txt-L1274-L1329
* I added some icons to the set to make things work: https://gist.github.com/TheSerapher/2602ca9ef656c2818b54b158a65da9e5#file-gistfile1-txt-L1148-L1159
* sensors are here: https://gist.github.com/TheSerapher/2602ca9ef656c2818b54b158a65da9e5#file-gistfile1-txt-L653-L694

Icons and texts have been renamed to match their sizes and some others added since I needed pretty small symbols.

There is also a second page `solar_page_2` which can be flipped to via the left and right arrows in the bottom left corner. Filled those with some additional consumers, replaced personalised names with generic `1`, `2`, `3`.

Thanks a lot for your work on this project @BigBobbas 🙂 I am really happy with the result and your code with plenty of examples made it a lot easier for me to understand the entire process!

**Comment by BigBobbas** [comment]
Looking good , Thank you so much for the "coffee" !! it reall means a lot and makes all the work worthwhile. 🙏 ❤️ 
Thank you for sharing this , i'm sure it will help someone out along the way.
Thank you for your support.
Best Wishes
Rob
